sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InProcessEvaluationContext.java

```diff
@@ -320,18 +320,31 @@ public String getStepName(AppliedPTransform<?, ?, ?> application) {
   }
 
   /**
-   * Returns a {@link SideInputReader} capable of reading the provided
+   * Returns a {@link ReadyCheckingSideInputReader} capable of reading the provided
    * {@link PCollectionView PCollectionViews}.
+   *
    * @param sideInputs the {@link PCollectionView PCollectionViews} the result should be able to
-   *                   read
-   * @return a {@link SideInputReader} that can read all of the provided
-   *         {@link PCollectionView PCollectionViews}
+   * read
+   * @return a {@link SideInputReader} that can read all of the provided {@link PCollectionView
+   * PCollectionViews}
    */
-  public SideInputReader createSideInputReader(final List<PCollectionView<?>> sideInputs) {
+  public ReadyCheckingSideInputReader createSideInputReader(
+      final List<PCollectionView<?>> sideInputs) {
     return sideInputContainer.createReaderForViews(sideInputs);
   }
 
   /**
+   * A {@link SideInputReader} that allows callers to check to see if a {@link PCollectionView} has
+   * had its contents set in a window.
+   */
+  static interface ReadyCheckingSideInputReader extends SideInputReader {
```

Comment:
This interface seems somewhat generally useful, particularly for the [recent suggestion](http://mail-archives.apache.org/mod_mbox/incubator-beam-dev/201604.mbox/%3CCANMXwW3n6j9AJ9NT4xGejdi%3DuSDOhvG%2BU3xeGv0fpHiK%2ByhhSQ%40mail.gmail.com%3E) by @aljoscha for a shared streaming side input infrastructure. Have you and Pei already discussed maybe putting it top level alongside `SideInputReader`?